coordinator/src/main/scala/filodb.coordinator/queryengine/FailureProvider.scala

```diff
@@ -0,0 +1,45 @@
+package filodb.coordinator.queryengine
+
+import filodb.query.LogicalPlan
+import filodb.query.exec.PlanDispatcher
+
+/**
+  * A provider to get failure ranges. Query engine can use failure ranges while preparing physical
+  * plan to reroute or skip failure ranges. Ranges are based on dataset and over all clusters.
+  */
+trait FailureProvider {
+  def getFailures(dataset: String, queryTimeRange: TimeRange): Seq[FailureTimeRange]
+}
+
+/**
+  * Time range.
+  * @param start epoch time in seconds.
+  * @param end epoch time in seconds.
+  */
+case class TimeRange(start: Long, end: Long)
+
+/**
+  * Failure details.
+  * @param pod cluster name.
+  * @param dataset dataset name
+  * @param timeRange time range.
+  * @param dispatcher dispatcher implementation for given pod.
+  */
+case class FailureTimeRange(pod: String, dataset: String, timeRange: TimeRange, dispatcher: PlanDispatcher)
```

Comment:
We may have to rename "pod" because of its numerous meanings. Lets discuss in person.

Reply:
Updated pod as clusterName.